### PR TITLE
Mde/simulation fix

### PR DESCRIPTION
**Pull request by dmarienko**
New position tracker's support based on stop or limit orders sent in advance

### CODE REVIEW

**Comment by yuriy-arabskyy** [comment]
several tests are failing for me on your branch, is it also happening for you?
Screenshot 2024-09-03 at 10 45 17

File: src/qubx/trackers/riskctrl.py [hunk 1]
```diff
@@ -219,7 +374,7 @@ def process_signals(self, ctx: StrategyContext, signals: List[Signal]) -> List[T
 
             target = self.get_position_sizer().calculate_target_positions(ctx, [s])[0]
             targets.append(target)
-            self._signals[s.instrument] = SgnCtrl(s, target, "NEW")
+            self._trackings[s.instrument] = SgnCtrl(s, target, "NEW")
```
Comment (yuriy-arabskyy) [inline src/qubx/trackers/riskctrl.py:377]: should "NEW" also be enum here?

Comment (dmarienko) [reply src/qubx/trackers/riskctrl.py:377]: Ohh, yep, my bad )), ty !

File: src/qubx/trackers/riskctrl.py [hunk 2]
```diff
@@ -95,37 +112,38 @@ def _get_price(update: float | Quote | Trade | Bar, direction: int) -> float:
...
+                        logger.debug(
+                            f"<yellow>{self.__class__.__name__}</yellow> triggered <red>STOP LOSS</red> for <green>{instrument.symbol}</green> at {c.signal.stop}"
+                        )
                         return TargetPosition.zero(
                             ctx,
                             instrument.signal(
                                 0,
                                 price=c.signal.stop,
                                 group="Risk Manager",
                                 comment="Stop triggered",
```
Comment (yuriy-arabskyy) [inline src/qubx/trackers/riskctrl.py:146]: this means that the basic stop taker tracker version will fill based on the market price right? and if we want exact fills that we need to use the advanced version.

Comment (dmarienko) [reply src/qubx/trackers/riskctrl.py:146]: Yes, I want to test it bit more today on my strategy and then switch to AdvancedStopTakeTracker
One thing I didn't like here - that we use inheritance for AtrStopTake etc 
I think we need to use decorator pattern here and pass instance of basic logic as parameter

Comment (yuriy-arabskyy) [reply src/qubx/trackers/riskctrl.py:146]: true, that would be better

**Review by yuriy-arabskyy — APPROVED** [review]

**Pull request merged**